sdk/simulation/tinst/t_instructions.cpp

```diff
@@ -207,6 +237,34 @@ static int _EGETKEY(sgx_key_request_t* kr, sgx_key_128bit_t okey)
         dd.ddpk.isv_prod_id = cur_secs->isv_prod_id;
         memcpy(&dd.ddpk.mrsigner, &cur_secs->mr_signer, sizeof(sgx_measurement_t));
         break;
+    case SGX_KEYSELECT_PROVISION_SEAL:
+        check_attr_flag(cur_secs, SGX_FLAGS_PROVISION_KEY);
+        check_isv_svn(kr, cur_secs);
+        check_config_svn(kr, cur_secs);
+        check_cpu_svn(kr);
+
+        // assemble derivation data
+        dd.size = sizeof(dd_provision_seal_key_t);
+        if (kr->key_policy & SGX_KEYPOLICY_ISVFAMILYID) {
+            memcpy(&dd.ddpsk.isv_family_id, &isv_ext_id->isv_family_id, sizeof(sgx_isvfamily_id_t));
+        }
+
+        if (kr->key_policy & SGX_KEYPOLICY_ISVEXTPRODID) {
+            memcpy(&dd.ddpsk.isv_ext_prod_id, &isv_ext_id->isv_ext_prod_id, sizeof(sgx_isvext_prod_id_t));
+        }
+
+        if (kr->key_policy & SGX_KEYPOLICY_CONFIGID) {
+            dd.ddpsk.config_svn = kr->config_svn;
+            memcpy(&dd.ddpsk.config_id, &cur_secs->config_id, sizeof(sgx_config_id_t));
+        }
+
+        memcpy(&dd.ddpsk.tmp_attr, &tmp_attr, sizeof(sgx_attributes_t));
+        memcpy(&dd.ddpsk.attribute_mask, &kr->attribute_mask, sizeof(sgx_attributes_t));
+        memcpy(&dd.ddpsk.cpu_svn,&kr->cpu_svn,sizeof(sgx_cpu_svn_t));
+        dd.ddpsk.isv_svn = kr->isv_svn;
+        dd.ddpsk.isv_prod_id = cur_secs->isv_prod_id;
```

Comment:
Also need to check if (!(kr->key_policy & SGX_KEYPOLICY_NOISVPRODID)) before set isv_prod_id.

Reply:
Done. Thanks for your review.